web/cypress/support/incidents_prometheus_query_mocks/prometheus-mocks.ts

```diff
@@ -0,0 +1,81 @@
+import { IncidentDefinition, PrometheusResponse, IncidentScenarioFixture } from './types';
+import { createIncidentMock, createAlertDetailsMock } from './mock-generators';
+import { convertFixtureToIncidents, parseYamlFixture } from './schema/fixture-converter';
+
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      mockIncidents(incidents: IncidentDefinition[]): Chainable<Element>;
+      mockIncidentFixture(fixturePath: string): Chainable<Element>;
+    }
+  }
+}
+
+/**
+ * Main mocking function - sets up cy.intercept for Prometheus query_range API
+ * Intercepts the query_range API and returns the mock data for the incidents
+ * @param incidents 
+ */
+export function mockPrometheusQueryRange(incidents: IncidentDefinition[]): void {
+  cy.intercept('GET', '/api/prometheus/api/v1/query_range*', (req) => {
+    const url = new URL(req.url, window.location.origin);
+    const query = url.searchParams.get('query') || '';
+
+    console.log(`INTERCEPTED: ${req.method} ${req.url}`);
+    console.log(`Query: ${query}`);
+
+    let results: any[]; 
+
+    if (query.includes('cluster:health:components:map')) {
```

Comment:
you can improve Readability with Early Return ,It's  clear what happens when the query isn't handled, and then the valid cases are processed afterward. Example
`if (!query.includes('cluster:health:components:map') && !query.includes('ALERTS{')) {
  console.log(`Passing through non-mocked query`);
  req.continue();
  return;
}

const results = query.includes('cluster:health:components:map')
  ? createIncidentMock(incidents, query)
  : createAlertDetailsMock(incidents, query);`